### PR TITLE
display rank in human readable + fix math

### PR DESCRIPTION
**Pull request by wonderingabout**
now using Math.floor instead of to.Fixed, this fixes the issue of rank miscalculation 

- now "minimum rank is 6k" for 5.3k
- now "maximum rank is 4d" for 4.8d

used this opportunity to rename botSecondsTime to humanReadableTimeString, for consistency with humanReadableMinRank and humanReadableMaxRank and humanReadableUserRank

also moved the timespanToDisplayString at the bottom, next to rankToString :+1:

### CODE REVIEW

File: connection.js [hunk 1]
```diff
@@ -311,13 +311,17 @@ class Connection {
...
+            return { reject: true, msg: "Minimum rank is " + humanReadableMinRank + " , your rank is too low, try again when your rank is high enough." };
         }
 
         if (user.ranking > config.maxrank) {
-            conn_log(user.username + " ranking too high: " + user.ranking);
-            return { reject: true, msg: "Maximum rank allowed is " + config.maxrank + " (in bot ranking units), your rank is " + user.ranking + " (in bot ranking units), your rank is too high, try again when your rank is low enough." };
+            let humanReadableUserRank = rankToString(user.ranking);
+            let humanReadableMaxRank = rankToString(config.maxrank);
+            conn_log(user.username + " ranking too high: " + humanReadableUserRank + " : max is " + humanReadableMaxRank);
+            return { reject: true, msg: "Maximum rank is " + humanReadableMaxRank + " , your rank is too high, try again when your rank is low enough." };
```
Comment (roy7) [inline connection.js]: Remove space before comma

File: connection.js [hunk 2]
```diff
@@ -311,13 +311,17 @@ class Connection {
         }
 
         if (user.ranking < config.minrank) {
-            conn_log(user.username + " ranking too low: " + user.ranking);
-            return { reject: true, msg: "Minimum rank allowed is " + config.minrank + " (in bot ranking units), your rank is " + user.ranking + " (in bot ranking units), your rank is too low, try again when your rank is high enough." };
+            let humanReadableUserRank = rankToString(user.ranking);
+            let humanReadableMinRank = rankToString(config.minrank);
+            conn_log(user.username + " ranking too low: " + humanReadableUserRank + " : min is " + humanReadableMinRank);
+            return { reject: true, msg: "Minimum rank is " + humanReadableMinRank + " , your rank is too low, try again when your rank is high enough." };
```
Comment (roy7) [inline connection.js]: Remove space before comma

File: connection.js [hunk 3]
```diff
@@ -449,9 +440,9 @@ class Connection {
...
                 t.main_time    < config.minmaintime) {  // others
-                    let botSecondsTime = timespanToDisplayString(config.minmaintime);
-                    conn_log(user.username + " wanted main time below minmaintime " + botSecondsTime);
-                    return { reject: true, msg: "Minimum main time is " + botSecondsTime + ", please increase main time \n - If you use canadian byo-yomi, set the average time per stone, for example 5 minutes/25 stones needs to be set up as 180 seconds (5 minutes) divided by 25 stones , which equals 7.2 seconds per stone. " };
+                    let humanReadableTimeString = timespanToDisplayString(config.minmaintime);
+                    conn_log(user.username + " wanted main time below minmaintime " + humanReadableTimeString);
+                    return { reject: true, msg: "Minimum main time is " + humanReadableTimeString + ", please increase main time \n - If you use canadian byo-yomi, set the average time per stone, for example 5 minutes/25 stones needs to be set up as 180 seconds (5 minutes) divided by 25 stones , which equals 7.2 seconds per stone. " };
```
Comment (roy7) [inline connection.js]: Remove space before comma

File: connection.js [hunk 4]
```diff
@@ -464,9 +455,9 @@ class Connection {
...
                 t.main_time    > config.maxmaintime) {  // others
-                    let botSecondsTime = timespanToDisplayString(config.maxmaintime);
-                    conn_log(user.username + " wanted main time above maxmaintime " + botSecondsTime);
-                    return { reject: true, msg: "Maximum main time is " + botSecondsTime + ", please reduce main time. \n - If you use canadian byo-yomi, set the average time per stone, for example 5 minutes/25 stones needs to be set up as 180 seconds (5 minutes) divided by 25 stones , which equals 7.2 seconds per stone. " };
+                    let humanReadableTimeString = timespanToDisplayString(config.maxmaintime);
+                    conn_log(user.username + " wanted main time above maxmaintime " + humanReadableTimeString);
+                    return { reject: true, msg: "Maximum main time is " + humanReadableTimeString + ", please reduce main time. \n - If you use canadian byo-yomi, set the average time per stone, for example 5 minutes/25 stones needs to be set up as 180 seconds (5 minutes) divided by 25 stones , which equals 7.2 seconds per stone. " };
```
Comment (roy7) [inline connection.js]: Remove space before comma

File: connection.js [hunk 5]
```diff
@@ -479,9 +470,9 @@ class Connection {
...
-                    let botSecondsTime = timespanToDisplayString(config.minmaintimeranked);
-                    conn_log(user.username + " wanted main time ranked below minmaintimeranked " + botSecondsTime);
-                    return { reject: true, msg: "Minimum main time for ranked games is " + botSecondsTime + ", please increase main time. \n - If you use canadian byo-yomi, set the average time per stone, for example 5 minutes/25 stones needs to be set up as 180 seconds (5 minutes) divided by 25 stones , which equals 7.2 seconds per stone. " };
+                    let humanReadableTimeString = timespanToDisplayString(config.minmaintimeranked);
+                    conn_log(user.username + " wanted main time ranked below minmaintimeranked " + humanReadableTimeString);
+                    return { reject: true, msg: "Minimum main time for ranked games is " + humanReadableTimeString + ", please increase main time. \n - If you use canadian byo-yomi, set the average time per stone, for example 5 minutes/25 stones needs to be set up as 180 seconds (5 minutes) divided by 25 stones , which equals 7.2 seconds per stone. " };
```
Comment (roy7) [inline connection.js]: Remove space before comma

File: connection.js [hunk 6]
```diff
@@ -494,9 +485,9 @@ class Connection {
...
-                    let botSecondsTime = timespanToDisplayString(config.maxmaintimeranked);
-                    conn_log(user.username + " wanted main time ranked above maxmaintimeranked " + botSecondsTime);
-                    return { reject: true, msg: "Maximum main time for ranked games is " + botSecondsTime + ", please reduce main time. \n - If you use canadian byo-yomi, set the average time per stone, for example 5 minutes/25 stones needs to be set up as 180 seconds (5 minutes) divided by 25 stones , which equals 7.2 seconds per stone. " };
+                    let humanReadableTimeString = timespanToDisplayString(config.maxmaintimeranked);
+                    conn_log(user.username + " wanted main time ranked above maxmaintimeranked " + humanReadableTimeString);
+                    return { reject: true, msg: "Maximum main time for ranked games is " + humanReadableTimeString + ", please reduce main time. \n - If you use canadian byo-yomi, set the average time per stone, for example 5 minutes/25 stones needs to be set up as 180 seconds (5 minutes) divided by 25 stones , which equals 7.2 seconds per stone. " };
```
Comment (roy7) [inline connection.js]: Remove space before comma

File: connection.js [hunk 7]
```diff
@@ -509,9 +500,9 @@ class Connection {
...
-                    let botSecondsTime = timespanToDisplayString(config.minmaintimeunranked);
-                    conn_log(user.username + " wanted main time unranked below minmaintimeunranked " + botSecondsTime);
-                    return { reject: true, msg: "Minimum main time for unranked games is " + botSecondsTime + ", please increase main time. \n - If you use canadian byo-yomi, set the average time per stone, for example 5 minutes/25 stones needs to be set up as 180 seconds (5 minutes) divided by 25 stones , which equals 7.2 seconds per stone. " };
+                    let humanReadableTimeString = timespanToDisplayString(config.minmaintimeunranked);
+                    conn_log(user.username + " wanted main time unranked below minmaintimeunranked " + humanReadableTimeString);
+                    return { reject: true, msg: "Minimum main time for unranked games is " + humanReadableTimeString + ", please increase main time. \n - If you use canadian byo-yomi, set the average time per stone, for example 5 minutes/25 stones needs to be set up as 180 seconds (5 minutes) divided by 25 stones , which equals 7.2 seconds per stone. " };
```
Comment (roy7) [inline connection.js]: Remove space before comma

File: connection.js [hunk 8]
```diff
@@ -524,9 +515,9 @@ class Connection {
...
-                    let botSecondsTime = timespanToDisplayString(config.maxmaintimeunranked);
-                    conn_log(user.username + " wanted main time unranked above maxmaintimeunranked " + botSecondsTime);
-                    return { reject: true, msg: "Maximum main time for unranked games is " + botSecondsTime + ", please reduce main time. \n - If you use canadian byo-yomi, set the average time per stone, for example 5 minutes/25 stones needs to be set up as 180 seconds (5 minutes) divided by 25 stones , which equals 7.2 seconds per stone. " };
+                    let humanReadableTimeString = timespanToDisplayString(config.maxmaintimeunranked);
+                    conn_log(user.username + " wanted main time unranked above maxmaintimeunranked " + humanReadableTimeString);
+                    return { reject: true, msg: "Maximum main time for unranked games is " + humanReadableTimeString + ", please reduce main time. \n - If you use canadian byo-yomi, set the average time per stone, for example 5 minutes/25 stones needs to be set up as 180 seconds (5 minutes) divided by 25 stones , which equals 7.2 seconds per stone. " };
```
Comment (roy7) [inline connection.js]: Remove space before comma

File: connection.js (hunk 3 above)
Comment (wonderingabout) [reply connection.js]: oh, its ok, this one will get replaced in the UHMAEAT anyways

File: connection.js [hunk 9]
```diff
@@ -527,7 +518,7 @@ class Connection {
                     }
                 }
             }
-
+
```
Comment (roy7) [inline connection.js]: Looks like a stray tab here

File: connection.js [hunk 10]
```diff
@@ -609,8 +600,6 @@ class Connection {
 
         if (t.periods > config.maxperiodsunranked && !notification.ranked) {
             conn_log(user.username + " wanted too many periods unranked: " + t.periods);
-            return { reject: true, msg: "Maximum number of periods for unranked games is " + config.maxperiodsunranked + " , please reduce the number of periods" };
```
Comment (roy7) [inline connection.js]: These 2 lines shouldn't be deleted?

**Comment by wonderingabout** [comment]
continues here : 

https://github.com/online-go/gtp2ogs/pull/117/

**Pull request closed without merging**